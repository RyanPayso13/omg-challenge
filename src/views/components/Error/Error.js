import React from 'react';
import { Flex, Text, Button } from 'rebass';
import { OverlayWrap } from '../Styled';

const Error = ({ callback }) => {
    return (
        <OverlayWrap
            data-testid="error-container">
            <Flex
                flexDirection="column"
                alignItems="center"
                justifyContent="center"
                width="100%">
                <Text 
                    data-testid="error-message"
                    color="#627381"
                    mb={ 2 }>
                    There has been an error!
                </Text>
                {!!callback && 
                    <Button 
                        data-testid="close-cta"
                        border="1px solid"
                        borderColor="#2b6cb0"
                        color="#2b6cb0"
                        bg="white"
                        onClick={ callback }>
                        Close
                    </Button>
                }
            </Flex>
        </OverlayWrap>
    );
};

export default Error;