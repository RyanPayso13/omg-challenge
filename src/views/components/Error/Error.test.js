import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Error from './Error';

describe('<Error />', () => {

    it('should render', () => {
        const { container } = render(<Error />);
        expect(container.firstChild).toMatchSnapshot();    
    });

    it('should have an error message', () => {
        const { getByTestId } = render(<Error />);
        expect(getByTestId('error-message')).toHaveTextContent('There has been an error!'); 
    });

    it('should have a close button', () => {
        const { getByTestId } = render(<Error callback={ jest.fn() } />);
        expect(getByTestId('error-message')).toBeInTheDocument(); 
    });  
    
    it('should execute the callback prop', () => {
        const dispatch = jest.fn();
        const { getByTestId } = render(<Error callback={ dispatch } />);
        fireEvent.click(getByTestId('close-cta'));
        expect(dispatch).toHaveBeenCalled(); 
    });  

});
