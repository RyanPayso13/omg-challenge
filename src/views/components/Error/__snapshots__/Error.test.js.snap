// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render 1`] = `
<div
  class="sc-gZMcBi eSpJzV"
  data-testid="error-container"
>
  <div
    class="sc-bdVaJa sc-bwzfXH dJYYq"
    width="100%"
  >
    <div
      class="sc-bdVaJa sc-htpNat AOxAi"
      color="#627381"
      data-testid="error-message"
    >
      There has been an error!
    </div>
  </div>
</div>
`;
